src/components/DownloadAndShareComponents/downloadsAndShare.types.ts

```diff
@@ -61,49 +61,78 @@ export type PreselectedDownloadType = Exclude<
   "video"
 >;
 
+export const isPreselectedDownloadType = (
+  preselected: PreselectedDownloadType | PreselectedShareType | null,
+): preselected is PreselectedDownloadType => {
+  return (
+    preselectedDownloadType.safeParse(preselected).success &&
+    preselected !== "video"
+  );
+};
+
+export const isPreselectedShareType = (
+  preselected: PreselectedDownloadType | PreselectedShareType | null,
+): preselected is PreselectedShareType => {
+  return preselectedShareType.safeParse(preselected).success;
+};
+
 export type PreselectedShareType = Exclude<
   z.infer<typeof preselectedResourceType>,
   "slide deck" | "additional material"
 >;
 
 export type ShareResourceType = LessonShareSchema["type"];
 
-export const preselectedDownloadTypeMap: Record<
-  PreselectedDownloadType,
-  DownloadResourceType[] | "all"
-> = {
-  "slide deck": ["presentation"],
-  "starter quiz": ["intro-quiz-questions", "intro-quiz-answers"],
-  "exit quiz": ["exit-quiz-questions", "exit-quiz-answers"],
-  worksheet: ["worksheet-pdf", "worksheet-pptx"],
-  "additional material": ["supplementary-pdf", "supplementary-docx"],
-  all: "all",
-};
+type CombinedPreselectedTypeMap = Record<
+  PreselectedDownloadType | PreselectedShareType,
+  {
+    downloadType?: DownloadResourceType[] | "all";
+    shareType?: LessonShareSchema["type"][] | "all";
```

Comment:
we have the `ShareResourceType` above 

side note but this type file is feeling a bit wild, could do with a tidy up? not necessarily to do in this pr, but there seems to be a lot of overlap/duplication making it difficult to find the type you need

Reply:
I've had a go at reducing the duplication, created a schema file and moved some of the functions out of the types file.